controllers/storagecluster/external_resources.go

```diff
@@ -291,6 +301,42 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) createDefaultClaimFor410UpgradeSCs(instance *ocsv1.StorageCluster) error {
+	cephFsStorageClassName := generateNameForCephFilesystemSC(instance)
+	cephRbdStorageClassName := generateNameForCephBlockPoolSC(instance)
+
+	cephFsStorageClass := &storagev1.StorageClass{}
+	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: cephFsStorageClassName}, cephFsStorageClass); err == nil {
+		err = r.createDefaultStorageClassClaimsForCephFS(instance)
+		if err != nil {
+			return fmt.Errorf("failed to created sharedfilesystem storageClassClaim %s. %v", cephFsStorageClassName, err)
+		}
+	}
+
+	cephRbdStorageClass := &storagev1.StorageClass{}
+	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: cephRbdStorageClassName}, cephRbdStorageClass); err == nil {
+		err := r.createDefaultStorageClassClaimsForRBD(instance)
+		if err != nil {
+			return fmt.Errorf("failed to created blockpool storageClassClaim %s. %v", cephRbdStorageClassName, err)
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) deleteSnapshotClassesFrom410(instance *ocsv1.StorageCluster) error {
+	snapshotClassConfiguration := newSnapshotClassConfigurations(instance)
+	for i := range snapshotClassConfiguration {
+		sc := snapshotClassConfiguration[i].snapshotClass
```

Comment:
This will create an in-memory copy of the snapshot class. suggesting:
```suggestion
		sc := &snapshotClassConfiguration[i].snapshotClass
```

Reply:
done

Reply:
Ok, I missed the fact that sc is already a pointer.
Resolving.